Stratis.Bitcoin/MemoryPool/MempoolPersistence.cs

```diff
@@ -0,0 +1,227 @@
+﻿using Stratis.Bitcoin.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NBitcoin;
+using System.IO;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Bitcoin.Logging;
+using Microsoft.Extensions.Logging;
+
+namespace Stratis.Bitcoin.MemoryPool
+{
+    public interface IMempoolPersistence
+    {
+        MemPoolSaveResult Save(TxMempool memPool, string fileName = null);
+        IEnumerable<MempoolPersistenceEntry> Load(string fileName = null);
+    }
+
+    public struct MemPoolSaveResult
+    {
+        public static MemPoolSaveResult NonSuccess { get { return new MemPoolSaveResult { Succeeded = false }; } }
+        public static MemPoolSaveResult Success(uint trxSaved) { return new MemPoolSaveResult { Succeeded = true, TrxSaved = trxSaved }; }
+
+        public bool Succeeded { get; private set; }
+        public uint TrxSaved { get; private set; }
+    }
+
+    public class MempoolPersistenceEntry : IBitcoinSerializable
+    {
+        byte[] tx;
```

Comment:
Instead of the byte[]? I can recode it that way if you want.
Still need this model though because feeDelta (priority) is captured from the txpool entry, it's not part of the Transaction itself